### PR TITLE
feat: add a new dynamic page to render the right hustler by id

### PR DESCRIPTION
**Pull request by smakosh**
## To Do
- [x] Create a new page at /hustlers/<id>/customize for updating the hustlers configuration.
- [x] We can reuse the existing customization components.
- [x] Fetch all metadata:
  - [x] sex
  - [x] bgColor
  - [x] textColor
  - [x] name
  - [x] facialHair
  - [x] hair
  - [x] zoomWindow
  - [x] bodyParts 
  - [x] options 
  - [x] mask
- [x] Override fetched metadata with new options when user customizes the hustler

> If the user is minting a new one, it should use the L1 mintTo, if they are customizing, it should use the L2 hustlers contract with setMetadata
- [x] The existing configuration can be pulled from: https://github.com/dopedao/dope-monorepo/blob/master/packages/contracts/dist/Hustler.d.ts#L335 (but i'll update the subgraph to pull the metadata too)
- [x] And updated with: https://github.com/dopedao/dope-monorepo/blob/master/packages/contracts/dist/Hustler.d.ts#L383

### CODE REVIEW

**Review by subimage — CHANGES_REQUESTED** [review]
@smakosh Looks like this is still in progress but here are the things I noticed…

[] On main net clicking my hustler brought up a random ID, not the hustler I wanted to configure
[] Configuration controls don't seem to work…

Let me know how I can help move this forward.

**Comment by smakosh** [comment]
> @smakosh Looks like this is still in progress but here are the things I noticed…
> 
> [] On main net clicking my hustler brought up a random ID, not the hustler I wanted to configure [] Configuration controls don't seem to work…
> 
> Let me know how I can help move this forward.

These have been now fixed

File: packages/web/components/hustler/RenderFromItemIds.tsx [hunk 1]
```diff
@@ -93,7 +93,7 @@ const RenderFromItemIds = ({
       const drugShadowHex = '0x00362f3729062b';
       hustlers
         .render(
-          renderName ? 'OG TITLE GOES HERE' : '', // title
+          renderName ? name : '', // title
```
Comment (tarrencev) [inline packages/web/components/hustler/RenderFromItemIds.tsx:96]: we should get the ogs title from the metadata

File: packages/web/hooks/contracts.ts [hunk 2]
```diff
@@ -65,13 +66,142 @@ export const useHustler = () => {
...
-  const initator = useInitiator();
+export const useFetchMetadata = () => {
+  const hustler = useHustler();
+  const { provider } = useOptimism();
 
-  useEffect(() => {
-    initator.release().then(date => setReleaseDate(new Date(date.toNumber() * 1000)));
-  }, [initator]);
+  return async function fetchMetadata(id: BigNumber) {
+    const metadata = await hustler.metadata(id);
```
Comment (tarrencev) [inline packages/web/hooks/contracts.ts:74]: we can get the OGs title off the Components contract for now: https://github.com/dopedao/dope-monorepo/blob/master/packages/contracts/src/Components.sol#L386

File: packages/web/features/hustlers/components/ConfigureHustler/index.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+      setname,
+      color,
+      background,
+      options,
+      zoomWindow,
+      bodyParts,
+      mask,
+    );
+
+    router.replace('/hustlers/mint-success');
```
[the remark this answers is not in the document]
Comment (smakosh) [reply packages/web/features/hustlers/components/ConfigureHustler/index.tsx:109]: What do you suggest we do once you're done customizing your hustler?

**Comment by smakosh** [comment]
Closing in favor of https://github.com/dopedao/dope-monorepo/pull/136

**Pull request closed without merging**